monkey/agent_plugins/exploiters/wmi/src/plugin.py

```diff
@@ -0,0 +1,120 @@
+import logging
+from functools import partial
+from pprint import pformat
+from typing import Any, Dict, Sequence
+from uuid import UUID
+
+# common imports
+from common.event_queue import IAgentEventPublisher
+from common.types import Event
+from common.utils.code_utils import del_key
+
+# dependencies to get rid of or internalize
+from infection_monkey.exploit import IAgentBinaryRepository, IAgentOTPProvider
+from infection_monkey.exploit.tools import (
+    BruteForceCredentialsProvider,
+    BruteForceExploiter,
+    generate_brute_force_credentials,
+)
+from infection_monkey.exploit.tools.helpers import get_agent_dst_path
+from infection_monkey.i_puppet import ExploiterResultData, TargetHost
+from infection_monkey.propagation_credentials_repository import IPropagationCredentialsRepository
+
+from .wmi_command_builder import build_wmi_command
+from .wmi_options import WMIOptions
+
+WMI_PORTS = [135]
+
+
+def should_attempt_exploit(host: TargetHost) -> bool:
+    closed_tcp_ports = host.ports_status.tcp_ports.closed
+    return not all([p in closed_tcp_ports for p in WMI_PORTS])
+
+
+logger = logging.getLogger(__name__)
+
+
+class PlaceholderWMIExploitClientFactory:
+    def __init__(self, *args, **kwargs):
+        pass
+
+
+class Plugin:
+    def __init__(
+        self,
+        *,
+        plugin_name: str,
+        agent_id: UUID,
+        agent_event_publisher: IAgentEventPublisher,
+        agent_binary_repository: IAgentBinaryRepository,
+        propagation_credentials_repository: IPropagationCredentialsRepository,
+        otp_provider: IAgentOTPProvider,
+        **kwargs,
+    ):
+        self._plugin_name = plugin_name
+        self._agent_id = agent_id
+        self._agent_event_publisher = agent_event_publisher
+        self._agent_binary_repository = agent_binary_repository
+        self._credentials_provider = BruteForceCredentialsProvider(
+            propagation_credentials_repository, generate_brute_force_credentials
```

Comment:
We should filter secret types by using `sercet_type_filter` and passing it to `generate_brute_force_credentials` with `functools.partial`